packages/react-next/src/components/Pivot/useRefEffect.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+/**
+ * Creates a ref, and calls a callback whenever the ref changes to a non-null value.
+ * The callback can optionally return a cleanup function that'll be called before the value changes or is set to null.
+ *
+ * This can be used to work around a limitation that useEffect cannot depend on `ref.current` (see
+ * https://github.com/facebook/react/issues/14387#issuecomment-503616820).
+ *
+ * @param callback - Called whenever the ref's value changes to non-null. Can optionally return a cleanup function.
+ * @param initialValue - (Optional) The initial value for the ref.
+ *
+ * @returns A tuple of `[ref, setRef]`:
+ * * `ref` - Read-only view of the ref's current value
+ * * `setRef` - A ref setter function that can be used to change the ref's current value.
+ */
+export function useRefEffect<T>(

```

Comment:
I like it :)

let's put this in `react-hooks` package? would be awesome if you can add unit test too

Reply:
Feel free to do this in a follow-up PR.

Reply:
Yeah, I was planning to move it to react-hooks in a future PR. Also to potentially move useOverflow to react-hooks as well, and observeResize to utilities.